thavalon-webapp/src/utils/GameSocket.tsx

```diff
@@ -14,39 +14,81 @@ export class GameSocket {
     // can also be undefined since it's not explicitly set in constructor
     private websocket: WebSocket;
 
-    // create the underlying websocket in the constructor
+    /**
+     * Construct the underlying websocket instance and set up function handlers.
+     * @param socketUrl 
+     */
     private constructor(socketUrl: string) {
         this.websocket = new WebSocket(socketUrl);
         this.websocket.onopen = this.socketOnOpen.bind(this);
         this.websocket.onmessage = this.socketOnMessage.bind(this);
         this.websocket.onclose = this.socketOnClose.bind(this);
         this.websocket.onerror = this.socketOnError.bind(this);
     } 
-    // functions for handling incoming websocket events
+
+    /**
+     * Listen to websocket's onopen event.
+     * @param event Event received when socket open.
+     */
     private socketOnOpen(event: Event) {
         console.log("Successfully initiated connection.");
     }
 
+    /**
+     * Listen to websocket's onmessage event.
+     * @param event Event received when socket gets message.
+     */
     private socketOnMessage(event: MessageEvent) {
         console.log(event);
         console.log("Received message: " + event.data);
     }
 
+    /**
+     * Listen to websocket's onclose event.
+     * @param event Event received when socket is closed.
+     */
     private socketOnClose(event: CloseEvent) {
         console.log("Recieved on close message.");
     }
 
+    /**
+     * Listen to websocket's onerror event.
+     * @param event Event received when socket errors.
+     */
     private socketOnError(event: Event) {
         console.log("Received on error message.");
     }
 
+    /**
+     * Send a message on the websocket. This will wait until websocket is
+     * open before sending the message.
+     * @param msg The message to be sent to the server, as a JSONified string. 
+     */
+    private sendMessage(msg: string) {
+        if (this.websocket.readyState === WebSocket.OPEN) {

```

Comment:
Should sendMessage accept an object instead of a string? Then we only need to serialize once here instead of having the caller serialize everything.

Reply:
Also, is it worth having an enum of OutboundMessageTypes to match the server?